### PR TITLE
*: tikv.NewMockTikvStore API refactor

### PR DESCRIPTION
**Pull request by tiancaiamao**
```
store, err := tikv.NewMockTikvStore(
	WithCluster(s.cluster),
	WithMVCCStore(s.mvccStore),
        WithHijackClient(wrap),
)
```

It looks much better.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @zimulala

File: store/tikv/kv.go [hunk 1]
```diff
@@ -137,33 +137,79 @@ func (s *tikvStore) EtcdAddrs() []string {
...
+	cluster        *mocktikv.Cluster
+	mvccStore      *mocktikv.MvccStore
+	clientHijack   func(Client) Client
+	pdClientHijack func(pd.Client) pd.Client
+}
+
+// MockTiKVStoreOption is used to control some behavior of mock tikv.
+type MockTiKVStoreOption func(*mockOptions)
+
+// WithHijackClient hijacks kv client's behavior, makes it easy to simulate the network
```
Comment (zimulala) [inline store/tikv/kv.go]: s/kv/KV

File: store/tikv/kv.go [hunk 2]
```diff
@@ -137,33 +137,79 @@ func (s *tikvStore) EtcdAddrs() []string {
...
+// problem between TiDB and TiKV.
+func WithHijackClient(wrap func(Client) Client) MockTiKVStoreOption {
+	return func(c *mockOptions) {
+		c.clientHijack = wrap
+	}
+}
+
+// WithHijackPDClient hijacks pd client's behavior, makes it easy to simulate the network
```
Comment (zimulala) [inline store/tikv/kv.go]: s/pd/PD

File: executor/executor_test.go [hunk 3]
```diff
@@ -63,7 +66,13 @@ func (s *testSuite) SetUpSuite(c *C) {
 	flag.Lookup("mockTikv")
 	useMockTikv := *mockTikv
 	if useMockTikv {
-		store, err := tikv.NewMockTikvStore("")
+		s.cluster = mocktikv.NewCluster()
+		mocktikv.BootstrapWithSingleStore(s.cluster)
+		s.mvccStore = mocktikv.NewMvccStore()
```
Comment (zimulala) [inline executor/executor_test.go]: We can use the default `opts`. Can we remove these lines?

Comment (tiancaiamao) [inline executor/executor_test.go]: We can't use the default `opts`, distsql_test use `s.cluster` to test table split.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**